### ISSUE TITLE
If training data and validation data is the same, how come the losses differ?

### ISSUE THREAD

**Issue by carlthome** [fchollet/keras#3562]
I suppose I'm missing something obvious but the progress bar reports a much higher training loss than validation loss when the data is the same. Why is that? Shouldn't they be pretty much the same?

With split data (80/20, for example) everything looks fine, with a slightly worse validation loss than training loss, as expected.

**Comment by dolaameng** [comment]
Answer in [FAQ](https://keras.io/getting-started/faq/#why-is-the-training-loss-much-higher-than-the-testing-loss)

**Comment by carlthome** [comment]
@dolaameng, thanks. :+1: 

> [...]the training loss is the average of the losses over each batch of training data. Because your model is changing over time, the loss over the first batches of an epoch is generally higher than over the last batches. On the other hand, the testing loss for an epoch is computed using the model as it is at the end of the epoch, resulting in a lower loss.

I would have expected the final progress bar print (the one that includes the validation loss) to do a proper evaluation, at least as an option, but that would be a bit computationally wasteful perhaps.
